torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -95,16 +97,58 @@ class FullyShardedDataParallel(nn.Module):
             params and grads to be on same device to work with optimizer. This
             API is subject to change. Default is ``None`` in which case there
             will be no offloading.
+        auto_wrap_policy: (Optional [callable]):
+            A callable specifying a policy to recursively wrap layers with FSDP.
+            Note that this policy currently will only apply to child modules of
```

Comment:
> If we need more flexibility and run into situations where the outer module should also not be wrapped, we can have a workaround where we run the outer module like a normal nn.Module by providing a special mode for this.

Could you please elaborate on the above statement? What's missing for the outer module? Do you mean the parameters directly created from the outer module?

Reply:
Does it mean we allow users to pass an FSDP module into FSDP ctor?

Reply:
I meant that current `wrap` API supports something like:
```
with enable_wrap(policy):
    wrap(mod) # assume mod, outer-most module does not satisfy policy
```

then `mod` will stay as an nn.Module, but with this new API, we have like - 
```
mod = FSDP(mod, policy)
```
and `policy` is currently only applied to children. It is tricky to apply `policy` in this way to `mod` itself because it will still be FSDP, since we call the ctor. 

> Does it mean we allow users to pass an FSDP module into FSDP ctor?

We disallow this, there is the function `_check_wrapped` to guard against it

Reply:
in real use cases, users always need to do FSDP(auto_wrap(module)), that means we always need an outermost/root FSDP instance for a module. so it is good to always return a root FSDP instance in the new main API FSDP(module, auto_wrap_policy)

Reply:
@mrshenli Does this explanation sound good to you?

Reply:
I see, this makes sense to me. Thanks!